tensor_genn/algorithms/relu_ann.py

```diff
@@ -103,26 +107,27 @@ def evaluate(self, X, y=None):
 
         runtime = n_examples * self.single_example_time
         n = len(self.neuron_pops)
-        i = -1
-        n_correct = 0
-
-        while self.g_model.t < runtime:
-            if self.g_model.t >= self.single_example_time*(i+1):
-                # After example i -1,0,1,2,..
-                self.g_model.pull_var_from_device("if"+str(n-1),'SpikeNumber')
-                SpikeNumber_view = self.neuron_pops[-1].vars["SpikeNumber"].view
-                n_correct += (np.argmax(SpikeNumber_view)==y[i])
-                i += 1
-                # Before example i 0,1,2,3,..
-                for j in range(len(self.neuron_pops)):
-                    self.neuron_pops[j].vars["SpikeNumber"].view[:] = 0
-                    self.neuron_pops[j].vars["Vmem"].view[:] = random.uniform(self.Vres,self.Vthr)
+        
+        n_correct = 0    
+
+        for i in range(n_examples):
+            # Before simulation
+            for j, npop in enumerate(self.neuron_pops):
+                    npop.vars["SpikeNumber"].view[:] = 0
+                    npop.vars["Vmem"].view[:] = random.uniform(self.Vres,self.Vthr)
                     self.g_model.push_state_to_device("if"+str(j))
                 
-                self.current_source.vars['magnitude'].view[:] = X[i] / 100.
-                self.g_model.push_var_to_device("cs",'magnitude')
+            self.current_source.vars['magnitude'].view[:] = X[i] / 100.
+            self.g_model.push_var_to_device("cs",'magnitude')
+
+            # Run simulation
+            while self.g_model.t < self.single_example_time * (i+1):
```

Comment:
You could just convert ``self.single_example_time`` into timesteps by dividing by ``self.timestep`` and taking the ceiling  and then just call ``self.g_model.step_time()`` that many times

Reply:
Oh yes! That'll make it much more independent. I'll do that right away